src/Components/PaperEditor/TranscriptsContainer/TranscriptTabContent/TranscriptMenu.js

```diff
@@ -13,16 +13,27 @@ import { faHighlighter, faCog } from '@fortawesome/free-solid-svg-icons';
 const TranscriptMenu = (props) => {
   const labels = props.labels;
   const [ isLabelsListOpen, setIsLabelsListOpen ] = useState(true);
-  const handleClick = (annotations) => {
-    //setAnnotations
-    props.handleClick(annotations);
+
+  const updateSelectedLabel = (e, labelId) => {
+    const tempLabels = labels;
+    const previousActiveLabel = tempLabels.find((label) => {
+      return label.active;
+    });
+    if (previousActiveLabel) {
+      previousActiveLabel.active = false;
+    }
+    const activeLabel = tempLabels.find((label) => {
+      return label.id === labelId;
+    });
+    activeLabel.active = true;
+    props.setLabels(tempLabels);
   };
```

Comment:
Here you are adding a logic to set the parent components state. In a child component you shouldn't be dealing with how the parent component's state is set. You should move this code up to the parent component, and not pass `setLabels` at all. 

e.g. line 69 of your code
```
<Row ... onClick={ (e) => props.updateSelectedLabel(e, label.id) }>
```